src/gtimelog/tests/test_timelog.py

```diff
@@ -1023,6 +1023,41 @@ def test_split_category_no_task_just_category(self):
         self.assertEqual(sp('project: '), ('project', ''))
         self.assertEqual(sp('project:'), ('project', ''))
 
+    def test_sorted_grouped_time_collection(self):
+        # the unsorted list is nromally a TimeCollection but we fake it
+        unsorted_list = (  # list of (start-time, name, duration)
+            (10_000, 'BBB: b', 20),
+            (30_000, 'CCC: c', 10),
+            (20_000, 'Alone', 12),
+            (40_000, 'AAA: a', 15),
+        )
+        taskfile = self.write_file('tasks.txt', textwrap.dedent('''\
+            Alone
+            # comments are skipped
+            BBB: b
+            AAA: a
+            CCC: c
+        '''))
+        tasklist = TaskList(taskfile)
+        sorted_by = {
+            'start-time': ('BBB: b', 'Alone', 'CCC: c', 'AAA: a'),
+            'name': ('AAA: a', 'Alone', 'BBB: b', 'CCC: c'),
+            'duration': ('CCC: c', 'Alone', 'AAA: a', 'BBB: b'),
+            'task-list': ('Alone', 'BBB: b', 'AAA: a', 'CCC: c'),
+        }
+
+        def tc_sorted(method):
+            tc_key = TimeCollection._get_grouped_order_key
+            return tuple(x[1] for x in
```

Comment:
For readability purposes I'd prefer `[entry for start_time, entry, duration in sorted(...)]`